### PR TITLE
Sync the internal GPTL lib of SCORPIO with the latest version in E3SM

### LINKED ISSUE #453: Support large pids in GPTL
Ensure that the version of GPTL in SCORPIO supports large PIDs.

This is already fixed in the version of GPTL used in E3SM. The change required is to modify the check for the pid range from
```
GPTLget_memusage.c :  103
  if (pid > 999999) {
    fprintf (stderr, "get_memusage: pid %d is too large\n", pid);
    return -1;
  }
```
to
```
  if (pid <= 0) {
    fprintf (stderr, "get_memusage: pid %d is non-positive\n", pid);
    return -1;
  }
```

A better solution would be to get the latest version of GPTL from E3SM & add back all the custom SCORPIO fixes manually.

Note that E3SM builds use the E3SM version of GPTL and hence do not encounter this issue.

### PR DESCRIPTION
**Pull request by dqwu**
E3SM builds use the E3SM version of GPTL under share/timing, which
has more fixes including large PIDs support.

This PR updates SCORPIO to get the latest version of GPTL from E3SM
with some custom fixes preserved.

Fixes #453

### CODE REVIEW

File: src/gptl/gptl.c [hunk 1]
```diff
@@ -801,18 +1112,27 @@ int GPTLstart_handle (const char *name,  /* timer name */
...
+  /*
+  ** Recursion => increment depth in recursion and return.  We need to return
   ** because we don't want to restart the timer.  We want the reported time for
   ** the timer to reflect the outermost layer of recursion.
   */
 
   if (ptr && ptr->onflg) {
-#ifdef GPTL_ASSERT_ON_RECURSION
-    assert(0);
-#endif
```
Comment (jayeshkrishna) [inline src/gptl/gptl.c]: Why did we lose this change?

**Review by jayeshkrishna — CHANGES_REQUESTED** [review]

**Comment by jayeshkrishna** [comment]
Couple of things to do before this PR gets merged,

* Ensure that the timing output generated (for SCORPIO standalone test) is the same (or looks good) with the new change
* List all the changes in SCORPIO (most of the changes should be in cmake files) version of GPTL and ensure that its present in this new version of code

File: src/gptl/gptl.c (hunk 1 above)
Comment (dqwu) [reply src/gptl/gptl.c]: You are right, this lost change is in PR #351 and we need to reapply it. Will recheck other changes as well.

File: src/gptl/COPYING [hunk 2]
```diff
@@ -1,674 +1,17 @@
...
+included in all copies or substantial portions of the Software.  Any
+commercial use (including sale) of the software, and derivative development
+towards commercial use, requires written permission of the copyright
+holder. THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO
+EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
```
[the remark this answers is not in the document]
Comment (dqwu) [reply src/gptl/COPYING:17]: @jayeshkrishna The reason for this change can be found in https://github.com/E3SM-Project/E3SM/commit/cabec9b62662519990a8b7159e5235b66d9551fa

[the remark this answers is not in the document]
Comment (jayeshkrishna) [reply src/gptl/COPYING:17]: ok, thanks

**Review by jayeshkrishna — APPROVED** [review]

**Pull request merged**